packages/esm-consultation-workflow/src/dynamic-workflow/components/regimen-drug-order/regimen-drug-order-step-renderer.component.test.tsx

```diff
@@ -0,0 +1,194 @@
+      dispenseTypes: mockDispenseTypes,
+      isLoading: false,
+    });
+  });
+
+  it('renders without crashing', () => {
+    render(<RegimenDrugOrderStepRenderer {...defaultProps} />);
+    expect(screen.getByText(/Dados do regime|regimenData/i)).toBeInTheDocument();
+  });
+
+  it('shows regimen select', () => {
+    render(<RegimenDrugOrderStepRenderer {...defaultProps} />);
+    const selects = screen.getAllByRole('combobox');
+    expect(selects.some((sel) => sel.id === 'regimen-select')).toBe(true);
+  });
+
+  it('disables add medication button if no regimen is selected', () => {
+    render(<RegimenDrugOrderStepRenderer {...defaultProps} />);
+    const addButton = screen.getByRole('button', { name: /Adicionar Medicamento/i });
+    expect(addButton).toBeDisabled();
+  });
+
+  it('displays loading state when fetching data', () => {
+    // Override the mock to show loading state
+    jest.mocked(useRegimensHook.useRegimens).mockReturnValue({
+      regimens: [],
+      isLoading: true,
+      error: null,
+    });
+
+    render(<RegimenDrugOrderStepRenderer {...defaultProps} />);
+    const loadingOption = screen.getByText('Loading...');
+    expect(loadingOption).toBeInTheDocument();
+  });
+
+  it('displays error message when there is an error fetching regimens', () => {
+    // Override the mock to show error state
+    jest.mocked(useRegimensHook.useRegimens).mockReturnValue({
+      regimens: [],
+      isLoading: false,
+      error: new Error('Failed to fetch regimens'),
+    });
+
+    render(<RegimenDrugOrderStepRenderer {...defaultProps} />);
+    // Just verify that the component renders without crashing when there's an error
+    expect(screen.getByText(/Dados do regime|regimenData/i)).toBeInTheDocument();
+  });
+
+  it('enables add medication button when regimen is selected', async () => {
+    render(<RegimenDrugOrderStepRenderer {...defaultProps} />);
+
+    // Find and select a regimen using a more reliable selector
+    const regimenSelect = document.getElementById('regimen-select');
+    fireEvent.change(regimenSelect, { target: { value: 'regimen1-uuid' } });
+
+    // Check if the add medication button is enabled
+    await waitFor(() => {
+      const addButton = screen.getByRole('button', { name: /Adicionar Medicamento/i });
+      expect(addButton).not.toBeDisabled();
+    });
+  });
+
+  it('enables add medication button after regimen is selected', async () => {
+    render(<RegimenDrugOrderStepRenderer {...defaultProps} />);
+
+    // First select a regimen (required to enable the add button)
+    const regimenSelect = document.getElementById('regimen-select');
+    fireEvent.change(regimenSelect, { target: { value: 'regimen1-uuid' } });
+
+    // Verify the add button becomes enabled
+    await waitFor(() => {
+      const addButton = screen.getByRole('button', { name: /Adicionar Medicamento/i });
+      expect(addButton).not.toBeDisabled();
+    });
```

Comment:
Usando `findBy*` não precisamos de `waitFor` acho que fica mais clean assim.
```suggestion
    const addButton = await screen.findByRole('button', { name: /Adicionar Medicamento/i });
    expect(addButton).not.toBeDisabled();
```